apps/system/js/wifi.js

```diff
@@ -208,15 +273,8 @@
       var lock = SettingsListener.getSettingsLock();
       // Let's quietly turn off wifi if there is no wake lock and
       // the screen is off and we are not on a power source.
-      if (!Service.query('screenEnabled') && !battery.charging) {
-        if (!this.wifiEnabled && this._wakeLockManager.isHeld) {
-          lock.set({ 'wifi.enabled': true });
-          window.addEventListener('wifi-enabled', function() {
```

Comment:
This `if` block was added in bug 979130 and should have been removed by bug 1127226.